include/ignition/gui/PlottingInterface.hh

```diff
@@ -0,0 +1,203 @@
+
+  /// \return value of the field
+  public: double Value();
+
+  /// \brief Register a chart that plot that field
+  public: void AddChart(int _chart);
+
+  /// \brief UnRegister a chart from plotting that field
+  /// \return the count of registered charts
+  public: void RemoveChart(int _chart);
+
+  /// \brief size of registered charts
+  /// \return charts size
+  public: int ChartCount();
+
+  public: std::set<int> &Charts();
+};
+
+class Topic
+{
+  /// \brief topic name
+  private: std::string name;
+
+  /// \brief Plotting fields to update its values
+  private: std::map<std::string, Field*> fields;
+
+  /// \brief Constructor
+  public: explicit Topic(std::string _name);
+
+  /// \brief get topic name
+  public: std::string Name();
+
+  /// \brief Register a chart to a field
+  /// \param[in] _fieldPath model path to the field as an ID
+  public: void Register(std::string _fieldPath, int _chart);
+
+  /// \brief remove field from the topic
+  public: void UnRegister(std::string _fieldPath, int _chart);
+
+  /// \brief size of registered fields
+  /// \return fields size
+  public: int FieldCount();
+
+  /// \brief get the registered topics
+  /// \return topics list
+  public: std::map<std::string, Field*> &Fields();
+  /// \brief callback to subscribe to that topic
+  /// \param[in] _msg the published msg from the topic
+  public: void Callback(const google::protobuf::Message &_msg);
+
+  /// \brief check the plotable types and get data from reflection
+  private: double PlotData(const google::protobuf::Message &_msg,
+                           const google::protobuf::FieldDescriptor *field);
+};
+
+class TransportPrivate;
+
+/// \brief handle Transport Topics Subscribing for one object(Chart)
+class Transport
+{
+  /// \brief Constructor
+  public: Transport();
+
+  /// \brief Destructor
+  public: ~Transport();
+
+  /// \brief unsubscribe/deattatch a field from a certain chart
+  /// \param[in] _topic topic name
+  /// \param[in] _fieldPath field path ID
+  /// \param[in] _chart chart ID
+  public: void Unsubscribe(std::string _topic,
+                           std::string _fieldPath,
+                           int _chart);
+
+  /// \brief subscribe/attatch a field from a certain chart
+  /// \param[in] _topic topic name
+  /// \param[in] _fieldPath field path ID
+  /// \param[in] _chart chart ID
+  public: void Subscribe(std::string _topic,
+                         std::string _fieldPath,
+                         int _chart);
+
+  /// \brief is the topic exist in the transport network
+  /// \param[in] _topic topic name
+  /// \return True if found in the transport, False if not found
+  public: bool TopicFound(const std::string &_topic);
+
+  /// \brief get the registered topics
+  /// \return topics list
+  public: const std::map<std::string, Topic*> &Topics();
+
+  private: std::unique_ptr<TransportPrivate> dataPtr;
+};
+
+class PlottingIfacePrivate;
+
+class PlottingInterface : public QObject
```

Comment:
Please document this class. It's especially important to document it because it's being installed and any downstream user could use it.